### DISCUSSION TITLE
Using LDT output as LAI data source

### DISCUSSION THREAD

**Discussion by farhadhassani** [NASA-LIS/LISF#992] [General]
I tried to use LDT output as an LAI source in LIS simulation. Also, I am using LIS 7.2 with NoahMP.4.0.1.
When I print out the LAI value "LIS_lai(n)%tlai(I)" in "subroutine LIS_read_lai" (LIS_historyMod.F90), it prints out the same value as the  LDT output. 
However, When I print out the LAI value "NOAHMP401_struc(n)%noahmp401(t)%lai" in NoahMP401_main.F90 it only prints 0 for LAI if the model starts from "coldstart" or the same as in the restart file if the model starts from a restart file.

There might be a problem in the connection between "LIS_lai(n)%tlai(I)" and "NOAHMP401_struc(n)%noahmp401(t)%lai", which should be considered.

Thank you in advance for your consideration.

Regards,
Farhad Hassani
farhad.hassani@mavs.uta.edu

**Comment by bmcandr** [comment]
Hi @farhadhassani,

Thanks for your message. I have converted your Issue into this Discussion until we confirm that (1) this is a bug and (2) it still exists. Regarding point 2, we no longer provide active support for LIS v7.2 Public and there have been many changes to the codebase since its release. I therefore recommend updating to our latest public release (v7.3.2) as a first step.

**Reply by farhadhassani** [reply]
Hi Brendan,

Thank you for your reply, I have tested it with (LISF_public_release_7.3.3) but I am still facing the same problem.

Regards,
Farhad

**Reply by bmcandr** [reply]
Sorry to hear that @farhadhassani. I am not super knowledgable in this area so I'm going to tag in my colleague @dmocko on our LSM team.

**Reply by bmcandr** [reply]
Also, please share your ldt.config and lis.config files. These may help us diagnose the issue.

**Comment by farhadhassani** [comment]
Thank you for your help. The ldt.config and lis.config and LDT output is attached here.

Regards,
Farhad

[lis_input_TX_d01_gen_lai.zip](https://github.com/NASA-LIS/LISF/files/8128200/lis_input_TX_d01_gen_lai.zip)

**Reply by dmocko** [reply]
Hi @farhadhassani 

Thanks for sharing your config files.  I see that you are trying this Noah-MP-4.0.1 vegetation option:
     Noah-MP.4.0.1 dynamic vegetation option:                      8  # 9 options available

Option 8 for Noah-MP-4.0.1's dynamic vegetation option is:    off (use input LAI; calculate FVEG)

I was aware of this code issue, and my apologies for not realizing you were also encountering it until I saw your config files.

I have been working on a solution, and suggest that you edit these two files in LISF/lis/surfacemodels/land/noahmp.4.0.1:

1) noahmp_driver_401.F90

```
571,572c571,572
<   if ((dveg_opt.eq.1).or.(dveg_opt.eq.6).or.(dveg_opt.eq.7)) then
<     ! with dveg_opt==1/6/7, shdfac is fed directly to fveg
---
>   if ( dveg_opt == 1 ) then
>     ! with dveg_opt==1, shdfac is fed directly to fveg
```

2) NoahMP401_main.F90

```
34d33
<     use LIS_vegDataMod,    only : LIS_lai
47c46
<     integer              :: row, col, tid
---
>     integer              :: row, col
527,535c526
<             if ((tmp_dveg_opt.ge.7).and.(tmp_dveg_opt.le.9)) then
<                tid = LIS_surface(n,LIS_rc%lsm_index)%tile(t)%tile_id
<                tmp_lai          = LIS_lai(n)%tlai(tid)
<                if (tmp_lai.eq.LIS_rc%udef) then
<                   tmp_lai          = NOAHMP401_struc(n)%noahmp401(t)%lai
<                endif
<             else
<                tmp_lai          = NOAHMP401_struc(n)%noahmp401(t)%lai
<             endif
---
>             tmp_lai             = NOAHMP401_struc(n)%noahmp401(t)%lai

```
Replace the code indicated by ">" with the lines indicated by "<".  The line numbers of the differences may not exactly correspond to your LISF release, but they should help you find the lines.  Please edit the files, re-compile, and run.

As you can see, this will properly send the LAI from LIS to the Noah-MP-4.0.1 physics.

I have been working on a pull request to update/fix this code from all.  I need to finish it up and get it merged in for everyone.

Please let me know if this works for you or not.

Sorry for the trouble,

David

**Reply by farhadhassani** [reply]
Dear David,

Thank you for your reply and I really appreciate it. I edited the files and it works well.

Best Regards,
Farhad

**Comment by dmocko** [comment]
Hi @farhadhassani,

One other thing that I noticed is that after running LDT, you don't have valid greenness and LAI/SAI parameter values for every land point in your domain.

I would suggest that you run LDT again to re-generate your lis_input file, and then re-do your simulations.

Please add/change this config entries in your ldt.config file:

```
Greenness spatial transform:   average
Greenness fill option:         average
Greenness fill radius:         5
Greenness fill value:          0.5
Greenness maximum fill value:   0.95
Greenness minimum fill value:   0.05

LAI/SAI spatial transform: average
LAI/SAI fill option: average
LAI/SAI fill radius:     5
LAI maximum fill value:  4
LAI minimum fill value:  0
LAI fill value:          0.5
SAI fill value:          0.5
```

These are just suggested fill values; you should determine what's most appropriate for your simulation.  Check the MaskParamFill.log file after you run LDT to see what values were filled at which x/y grid points.  For the most part, LDT won't actually use the fill values in the config file, but will instead fill using the average of other grid points.  "Average" is appropriate, as you are going upscale from 0.01-deg to your running domain.

David

**Comment by farhadhassani** [comment]
Hi David

Thank you for your help, I will try to edit the ldt config and rerun the model.

Sincerely,
Farhad
